python/pyspark/sql/dataframe.py

```diff
@@ -6281,6 +6282,10 @@ def toPandas(self) -> "PandasDataFrameLike":
         """
         ...
 
+    @property
+    def queryExecution(self) -> Optional["QueryExecution"]:
```

Comment:
Should probably have docstring here, with the added version

Reply:
And I wouldn't make it `Optional`.

Reply:
I added the version, for the optional part:

- in scala QueryExecution is always present but then you have to check for `executedPlan` the thing I'm worried about is to bring this complexity to the client. The Query Execution object allows too much direct manipulation of the query that is not ideal.

Reply:
Does Scala client have `df.queryExecution`? I don't see it in the Scala client.
If not, I guess we should avoid the same name as it's confusing because it's missing many properties that exist in the classic's `df.queryExecution`.

I'd rather prefer `CollectedMetrics` and it can be added to the classic as well to be consistent.

Reply:
IC so it has to be set after execution. Should we probably have a Spark Connect dedicated API? I think it'd be confusing if it has the same name with Scala side `df.queryExecution`